sda-commons-server-auth-testing/src/main/java/org/sdase/commons/server/auth/testing/AuthRule.java

```diff
@@ -104,30 +107,38 @@ private void init() {
 
   private void initDisabledTestAuth() {
     this.authConfig = new AuthConfig().setDisableAuth(true);
-    delegate =
-        RuleChain.outerRule(
-            new SystemPropertyRule().setProperty(AUTH_RULE_ENV_KEY, "{\"disableAuth\": true}"));
+    initTestAuth("{\"disableAuth\": true}");
   }
 
   private void initEnabledTestAuth() {
     this.privateKey = loadPrivateKey(this.privateKeyLocation);
     this.authConfig = new AuthConfig().setKeys(singletonList(createKeyLocation()));
 
     try {
-      String authKeysConfig = new ObjectMapper().writeValueAsString(authConfig);
-      delegate =
-          RuleChain.outerRule(
-              new SystemPropertyRule().setProperty(AUTH_RULE_ENV_KEY, authKeysConfig));
+      final String authKeysConfig = new ObjectMapper().writeValueAsString(authConfig);
+      initTestAuth(authKeysConfig);
     } catch (JsonProcessingException e) {
       fail("Failed to create the config keys: " + e.getMessage());
     }
   }
 
+  private void initTestAuth(String authConfig) {
+    final TestRule testRule = createTestRule(authConfig);
+    delegate = RuleChain.outerRule(testRule);
+  }
+
+  public static TestRule createTestRule(String authConfig) {
```

Comment:
Could we use less visibility to make it internal or is anybody required to use this method apart from the internal tests?

Reply:
Package private should also work. Thanks for pointing it out! 👍